src/main/java/com/minecolonies/coremod/colony/buildings/modules/BuildingModules.java

```diff
@@ -513,7 +514,7 @@ public class BuildingModules
       new BuildingEntry.ModuleProducer<>("tool_scepterguard_view", null, () -> () -> new ToolModuleView(
         ModItems.scepterGuard));
     public static final BuildingEntry.ModuleProducer<SettingsModule,SettingsModuleView> GUARD_SETTINGS = new BuildingEntry.ModuleProducer<>("guard_settings", () -> new SettingsModule()
-      .with(AbstractBuildingGuards.GUARD_TASK, new GuardTaskSetting(GuardTaskSetting.PATROL, GuardTaskSetting.GUARD, GuardTaskSetting.FOLLOW))
+      .with(AbstractBuildingGuards.GUARD_TASK, new GuardTaskSetting(GuardTaskSetting.PATROL, GuardTaskSetting.GUARD, GuardTaskSetting.FOLLOW, GuardTaskSetting.PATROL_MINE))
```

Comment:
Firstly, if you want all the values then you can use the parameterless constructor.

Secondly, this will give barracks guards the ability to patrol the mine as well, which was not previously the case and is probably not intended (as you'd still only be able to assign a single guard at the mine itself and then the squad AI would probably get borked).  These settings were deliberately different between guard tower and barracks tower.